### DISCUSSION TITLE
Model recommendation for dataset

### DISCUSSION THREAD

**Discussion by TheDragon246** [pyg-team/pytorch_geometric#4651] [Q&A]
I am trying to predict a value from a graph that is created using MRI data, which each 3D point has 8 nearest points mentioned using knn_graph from torch_cluster and saved in edge_index. Every point also has n node features (n=64 on current data but is variable. It is basically 'parcellation' values per point when I take 64 parcellations for the data).

This is the data structure that I have currently -
`Data(edge_index=[2, 1049904], pos=[131238, 3], x=[131238, 64], y=0.547589624452)`
for 1 of 100 participants, so I have 100 of these in the same format.
1049904 is 131238 (number of points) times 8.

My question is if there is a speciifc model that is suitable to predict y from using node features and edge_index? As I understand many models that are implemented in geometric (like Pointnet++) are node classification in nature, while I may require graph classification(?) with data per participant in the format mentioned above, predicting the y value.

Are there any implemented models in geometric that would be useful for this use case? Any suggestions would be helpful. Thanks!

**Comment by rusty1s** [comment]
PointNet++ both support node-level and graph-level classification. The latter is probably a good baseline you can try out, see https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pointnet2_classification.py.
